File: packages/health_connector_hk_ios/ios/health_connector_hk_ios/Sources/health_connector_hk_ios/mappers/measurement_unit_mappers/TimeDurationMapper.swift
import Foundation
import HealthKit

extension TimeDurationDto {
    /// Converts this DTO to a TimeInterval (seconds as Double).
    func toTimeInterval() -> TimeInterval {
        TimeInterval(seconds)
    }
}

extension TimeInterval {
    /// Converts a TimeInterval (seconds) to an TimeDurationDto.
    ///
    /// Uses seconds as the transfer unit for consistency.
    func toIntervalDto() -> TimeDurationDto {
        TimeDurationDto(seconds: self)
    }
}

File: packages/health_connector_hc_android/lib/src/mappers/measurement_unit_mappers/time_duration_mapper.dart
import 'package:health_connector_core/health_connector_core_internal.dart'
    show TimeDuration;
import 'package:health_connector_hc_android/src/pigeon/health_connector_hc_android_api.g.dart'
    show TimeDurationDto;
import 'package:meta/meta.dart' show internal;

/// Converts [TimeDuration] to [TimeDurationDto].
@internal
extension TimeDurationToDto on TimeDuration {
  TimeDurationDto toDto() {
    // Uses seconds as the transfer unit for consistency.
    return TimeDurationDto(
      seconds: inSeconds,
    );
  }
}

/// Converts [TimeDurationDto] to [TimeDuration].
@internal
extension TimeDurationDtoToDomain on TimeDurationDto {
  TimeDuration toDomain() {
    return TimeDuration.seconds(seconds);
  }
}
